doc/rfc/5910-consistent-sql-lua-types.md

```diff
@@ -0,0 +1,1700 @@
+to comparisons with scalar `SCALAR` values, then we could get conflicting
+results given we used index or not. E.g.:
+
+```
+tarantool> box.execute('CREATE TABLE t (i SCALAR PRIMARY KEY, a SCALAR);')
+---
+- row_count: 1
+...
+
+tarantool> box.execute('INSERT INTO t VALUES (10, 10);')
+---
+- row_count: 1
+...
+
+tarantool> box.execute([[SELECT * FROM t WHERE i < '1';]])
+---
+- metadata:
+  - name: I
+    type: scalar
+  - name: A
+    type: scalar
+  rows:
+  - [10, 10]
+...
+
+tarantool> box.execute([[SELECT * FROM t WHERE a < '1';]])
+---
+- metadata:
+  - name: I
+    type: scalar
+  - name: A
+    type: scalar
+  rows: []
+...
+```
+
+* To avoid this problem we decided to __not use__ generic implicit conversion table
+  for comparisons, essentially disabling them, and use instead box' [comparison rules]
+  (#comparison-operators);
+
+* With special treatment of comparisons we do not need to disable implicit casts
+  al together for scalar columns (as it were suggested before by some one).
+  It's not cast was problem here, but different results from iterators.
+
+## Implicit casts
+
+Here is the older picture of implicit conversions that we do in vdbe
+code:
+
+<img src="media/image16.png" />
+
+"Integer" (actually negative integer) and "unsigned" make up the generic
+"integer" type. Number is a superset of generic integer and double.
+(There is no smaller 4-byte float data type, as in other SQL vendors'
+tables, because it does not make much sense for NoSQL storage).
+
+Here are a few comments about possible combinations in the above table:
+
+-   Depending on current value content of `any` typed column, it may be
+    conditionally converted to any other target type;
+
+-   The same conditional logic applies for `scalar` to any other
+    (scalar) target type.
+
+-   Integer (unsigned and signed) values might be implicitly converted
+    from strings (if passed in proper format), double (if conversion
+    would be lossless) or from another integer value. `number` is
+    supertype for integers and floats, thus these conditions are correct
+    for conversion from `number` as well;
+
+-   Similarly, conversion to `double` might be performed from `string`,
+    `integer`, another `float`, or `number`;
+
+-   `boolean`s can only be converted to `boolean`s;
+
+-   `varbinary` may be converted to `string` and `varbinary`. The reverse is
+    not true for string to varbinary conversion, it's not allowed for
+    implicit cast (but is allowed for _explicit_ cast);
+
+-   Aggregate types like `array` or `map` can only be converted to array or
+    map correspondingly. This is already in the code and does not
+    require any addition after native ARRAY/MAP support will be added to the
+    SQL parser;
+
+
+### Implicit types conversion refactoring
+
+There is refactoring going on for a stricter casts table, which suggests
+to implement this (slightly stricter) conversion table:
+
+<img src="media/image17.png" />
+
```

Comment:
> * Why UNSIGNED and INTEGER cannot be implicitly cast to BOOLEAN?
> 
> * Why STRING cannot be implicitly cast to BOOLEAN (For example only 'TRUE' and 'FALSE'). The same about VARBINARY.
> 
> * Why STRING cannot be implicitly cast to VARBINARY?
> 
> * Why BOOLEAN cannot be implicitly cast to STRING?
> 
> * Why NUMBER cannot be implicitly cast to STRING?

That's conversion table which we would end up, taking into account your branch which was removing casts, where I looked into. ¯\_(ツ)_/¯

But in any case, to reduce confusion we could get rid _entirely_ of this chapter with refactoring, because it looks 
very much confusing for us, and this table will not go into future version.

Reply:
Deleted, as not representing our current plan of actions.